ledger/alonzo/pparams.go

```diff
@@ -150,16 +167,78 @@ func (p *AlonzoProtocolParameters) UpdateFromGenesis(genesis *AlonzoGenesis) {
 		Memory: uint64(genesis.MaxBlockExUnits.Mem),
 		Steps:  uint64(genesis.MaxBlockExUnits.Steps),
 	}
-	if genesis.ExecutionPrices.Mem != nil &&
-		genesis.ExecutionPrices.Steps != nil {
+
+	if genesis.ExecutionPrices.Mem != nil && genesis.ExecutionPrices.Steps != nil {
 		p.ExecutionCosts = common.ExUnitPrice{
 			MemPrice:  &cbor.Rat{Rat: genesis.ExecutionPrices.Mem.Rat},
 			StepPrice: &cbor.Rat{Rat: genesis.ExecutionPrices.Steps.Rat},
 		}
 	}
-	// TODO: cost models (#852)
-	// We have 150+ string values to map to array indexes
-	//	CostModels           map[string]map[string]int
+
+	if genesis.CostModels != nil {
+		p.CostModels = make(map[uint][]int64)
+
+		for versionStr, model := range genesis.CostModels {
+			key, ok := plutusVersionToKey(versionStr)
+			if !ok {
+				continue
+			}
+
+			var values []int64
+			switch v := model.(type) {
+			case map[string]interface{}:
+				maxIndex := 0
+				// Find maximum parameter index
+				for paramName := range v {
+					var index int
+					if _, err := fmt.Sscanf(paramName, "param%d", &index); err == nil && index > maxIndex {
+						maxIndex = index
+					}
+				}
+				values = make([]int64, maxIndex)
+				for paramName, val := range v {
+					var index int
+					if _, err := fmt.Sscanf(paramName, "param%d", &index); err == nil && index > 0 {
+						if intVal, ok := val.(float64); ok {
+							values[index-1] = int64(intVal)
+						}
+					}
+				}
+
+			case []interface{}:
+				values = make([]int64, len(v))
+				for i, val := range v {
+					if intVal, ok := val.(float64); ok {
+						values[i] = int64(intVal)
+					}
```

Comment:
Will this only ever contain `float64` values? It looks like anything else would be ignored